pkg/server/server.go

```diff
@@ -66,12 +67,25 @@ func (s *CSIDriverProviderServer) Mount(ctx context.Context, req *v1alpha1.Mount
 	f := []*v1alpha1.File{}
 	// CSI driver v0.0.21+ will write to the filesystem if the files are in the response.
 	// No files in the response translates to "not implemented" in the CSI driver.
-	for k, v := range files {
-		f = append(f, &v1alpha1.File{
-			Path:     k,
-			Contents: v,
-			Mode:     int32(filePermission),
-		})
+	for _, file := range files {
+		secretFile := &v1alpha1.File{
+			Path:     file.Path,
+			Contents: file.Content,
+		}
+
+		if file.FileMode != "" {
+			permission, parseErr := strconv.ParseInt(file.FileMode, 8, 32)
```

Comment:
I think we should validate the file permission before fetching the secrets from keyvault to fail fast. This way we avoid making additional API calls when the final result is going to be an error.